src/cli/commands/upgrade.js

```diff
@@ -19,6 +20,20 @@ export async function run(
   args: Array<string>,
 ): Promise<void> {
   const lockfile = args.length ? await Lockfile.fromDirectory(config.cwd, reporter) : new Lockfile();
-  const install = new Add(args, flags, config, reporter, lockfile);
+  let addArgs = [...args];
+
+  if (addArgs.length) {
+    const [dependency] = addArgs;
```

Comment:
Just did some tests:

```
~/Projects/yarn (upgrade_git_packages *) $ yarn upgrade foo bar --ignore-index
yarn upgrade v0.17.4
yarn upgrade args: [ 'foo', 'bar' ]
```

I'll switch this out to iterate and appropriate build the new args.